File: web-ss/src/core/DTO/UserDTO.ts
import { Role } from '../ENUM/Role';

export class UserDTO {
    private _nome: string;
    private _sobrenome: string;
    private _login: string;
    private _datanascimento: Date;
    private _email: string;
    private _password: string;
    private _role: Role;
    constructor(nome: string, sobrenome: string, login: string, datanascimento: Date, email: string, password: string,
                role: Role) {
        this._nome = nome;
        this._sobrenome = sobrenome;
        this._login = login;
        this._datanascimento = datanascimento;
        this._email = email;
        this._password = password;
        this._role = role;
    }

    get nome(): string {
        return this._nome;
    }

    set nome(value: string) {
        this._nome = value;
    }

    get sobrenome(): string {
        return this._sobrenome;
    }

    set sobrenome(value: string) {
        this._sobrenome = value;
    }

    get login(): string {
        return this._login;
    }

    set login(value: string) {
        this._login = value;
    }

    get datanascimento(): Date {
        return this._datanascimento;
    }

    set datanascimento(value: Date) {
        this._datanascimento = value;
    }

    get email(): string {
        return this._email;
    }

    set email(value: string) {
        this._email = value;
    }

    get password(): string {
        return this._password;
    }

    set password(value: string) {
        this._password = value;
    }

    get role(): Role {
        return this._role;
    }

    set role(value: Role) {
        this._role = value;
    }
}

File: backend/src/main/java/com/smartSpd/smartSpding/Core/DTO/UserDTO.java
package com.smartSpd.smartSpding.Core.DTO;

import java.time.LocalDate;

public class UserDTO {
    private String nome;
    private String sobrenome;
    private String login;
    private LocalDate datanascimento;
    private String email;
    private String password;
    private String role;

    public UserDTO(String nome, String sobrenome, String login, LocalDate datanascimento,
                   String email, String password, String role) {
        this.nome = nome;
        this.sobrenome = sobrenome;
        this.login = login;
        this.datanascimento = datanascimento;
        this.email = email;
        this.password = password;
        this.role = role;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getSobrenome() {
        return sobrenome;
    }

    public void setSobrenome(String sobrenome) {
        this.sobrenome = sobrenome;
    }

    public String getLogin() {
        return login;
    }

    public void setLogin(String login) {
        this.login = login;
    }

    public LocalDate getDatanascimento() {
        return datanascimento;
    }

    public void setDatanascimento(LocalDate datanascimento) {
        this.datanascimento = datanascimento;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }
}
